lrauv_ignition_plugins/test/test_sensor.cc

```diff
@@ -99,58 +152,44 @@ TEST(SensorTest, Sensor)
   }
   ASSERT_LE(sleep, maxSleep);
 
-  double spawnDepth{50};
 
-  // No specific orientation, vehicle will face North
-  ignition::math::Angle lat1 = IGN_DTOR(36.7999992370605);
-  ignition::math::Angle lon1 = IGN_DTOR(-122.720001220703);
-  {
-    lrauv_ignition_plugins::msgs::LRAUVInit spawnMsg;
-    spawnMsg.mutable_id_()->set_data("vehicle1");
-    spawnMsg.set_initlat_(lat1.Degree());
-    spawnMsg.set_initlon_(lon1.Degree());
-    spawnMsg.set_initz_(spawnDepth);
-    spawnMsg.set_acommsaddress_(201);
-
-    spawnPub.Publish(spawnMsg);
-  }
+  // On a depth coplanar to a piece of data
+  SpawnVehicle(
+    spawnPub, "vehicle1", 36.7999992370605, -122.720001220703, 50, 0);
+  node.Subscribe("/model/vehicle1/chlorophyll", &ChlorophyllVeh1Cb);
+
+  // At the surface should still have reading
+  SpawnVehicle(
+    spawnPub, "vehicle2", 36.7999992370605, -122.720001220703, 0, 0);
+  node.Subscribe("/model/vehicle2/chlorophyll", &ChlorophyllVeh2Cb);
+
+  // In the middle of two slices, still in a plane
+  SpawnVehicle(
+    spawnPub, "vehicle3", 36.7999992370605, -122.720001220703, 45, 0);
+  node.Subscribe("/model/vehicle3/chlorophyll", &ChlorophyllVeh3Cb);
+
+  // In the middle of a quadrant
+  SpawnVehicle(
+    spawnPub, "vehicle4", 36.7999992370605, -122.7, 48, 0);
+  node.Subscribe("/model/vehicle4/chlorophyll", &ChlorophyllVeh4Cb);
 
   // Check that vehicle was spawned
   int expectedIterations = iterations;
-  for (sleep = 0; ignition::gazebo::kNullEntity == vehicle1 && sleep < maxSleep;
+  for (sleep = 0; spawnedAllVehicles.load() == false && sleep < maxSleep;
```

Comment:
@arjo129 nit^2: 
```suggestion
  for (sleep = 0; !spawnedAllVehicles.load() && sleep < maxSleep;
```
is equivalent.